### PR TITLE
Modularize content

### PR DESCRIPTION
**Pull request by mavam**
This PR makes it easier to embed the content of this repository in the official documentation by splitting the content into seperate files.

### CODE REVIEW

File: coding-style.md [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+style][google-style], and [CAF style][caf-style] guidelines.
+
+[google-style]: http://google-styleguide.googlecode.com/svn/trunk/cppguide.xml
+[caf-style]: https://github.com/actor-framework/actor-framework/blob/master/CONTRIBUTING.md
+
+### General
+
+- Use 2 spaces per indentation level.
+
+- No tabs, ever.
```
Comment (dominiklohmann) [inline coding-style.md]: Both of these are enforced by clang-format; no need to mention them explicitly.

File: coding-style.md [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+
+- No C-style casts, ever.
+
+- 80 characters max per line.
```
Comment (dominiklohmann) [inline coding-style.md]: Enforced by clang-format.

File: coding-style.md [hunk 3]
```diff
@@ -0,0 +1,552 @@
...
+
+- Minimize vertical whitespace within functions. Use comments to separate
+  logical code blocks.
+
+- Namespaces and access modifiers (e.g., `public`) do not increase the
+  indentation level.
```
Comment (dominiklohmann) [inline coding-style.md]: Enforced by clang-format.

File: coding-style.md [hunk 4]
```diff
@@ -0,0 +1,552 @@
...
+
+- The `const` keyword precedes the type, e.g., `const T&` as opposed to
+  `T const&`.
```
Comment (dominiklohmann) [inline coding-style.md:100]: cc @lava on east vs. west const: We actually have this in here

File: coding-style.md [hunk 5]
```diff
@@ -0,0 +1,552 @@
...
+  structure includes in unit tests.
+
+- As in the standard library, the order of parameters when declaring a function
+  is: inputs, then outputs. API coherence and symmetry trumps this rule, e.g.,
+  when the first argument of related functions model the same concept.
+
+### Classes
+
+- Use the order `public`, `protected`, `private` for functions and members in
+  classes.
```
Comment (dominiklohmann) [inline coding-style.md:183]: This is sometimes not enforceable for classes with inline function definitions.

File: coding-style.md [hunk 6]
```diff
@@ -0,0 +1,552 @@
...
+
+- Mark single-argument constructors as `explicit` to avoid implicit
+  conversions.
```
Comment (dominiklohmann) [inline coding-style.md]: We should add to use `explicit(false)` if a constructor being implicit is actually intended so it's clear that it's not a bug.

File: coding-style.md [hunk 7]
```diff
@@ -0,0 +1,552 @@
...
+  template <class T>
+  using my_trait_t = typename my_trait<T>::type;
+
+  template <class T>
+  constexpr auto my_trait_v = my_trait<T>::value;
+  ```
+
+### Logging
+
+- Available log levels are *ERROR*, *WARNING*, *INFO*, *DEBUG* and *TRACE*.
```
Comment (dominiklohmann) [inline coding-style.md]:
```suggestion
- Available log levels are *ERROR*, *WARN*, *INFO*, *VERBOSE*, *DEBUG* and *TRACE*.
```

File: coding-style.md [hunk 8]
```diff
@@ -0,0 +1,552 @@
...
+
+- Messages can be sent by using the `VAST_\<level>` and `VAST_\<level>_ANON`
+  macros.
```
Comment (dominiklohmann) [inline coding-style.md]:
```suggestion
- Messages can be sent by using the `VAST_\<level>` macros.
```

File: coding-style.md [hunk 9]
```diff
@@ -0,0 +1,552 @@
...
+  index got a request for unkown query ID <181f5b1f-c673-4e98-acd2-ef762fe567a2>
+  ```
+
+- The `VAST_\<level>_ANON` macros are available for situations where **n**o
+  **s**ubject is suitable for the first argument to the other macro variant.
+  They can either be used to inject a subject manually, or create a message
+  in the *gerund* form. For example:
+  ```cpp
+  VAST_DEBUG_ANON(__func__, "creating directory", dir);
+  ```
```
Comment (dominiklohmann) [inline coding-style.md]: This is wholly outdated.

File: coding-style.md [hunk 10]
```diff
@@ -0,0 +1,552 @@
...
+
+- By default, use the `VAST_\<level>` variants.
```
Comment (dominiklohmann) [inline coding-style.md]: This no longer applies as the ANON variant doesn't exist anymore.

File: coding-style.md [hunk 11]
```diff
@@ -0,0 +1,552 @@
...
+- Try to restrict usage of the `VAST_INFO` message type to the main actors.
+  Info is the chattiest level that most users will see, so it should require
+  no or only little understanding of VASTs system architecture for the
+  reader to understand.
+
+- Messages sent at the trace level add the additional effect of writing a
+  second message at the exit of the current scope. The trace level can be
+  used to create a trace of the call stack with fine grained control over
+  its depth. Omit trace messages from helper functions and generic / general
+  purpose algorithm implementations.
```
Comment (dominiklohmann) [inline coding-style.md]: This was moved into a separate VAST_TRACE_SCOPE macro.

File: coding-style.md [hunk 12]
```diff
@@ -0,0 +1,552 @@
...
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
```
Comment (dominiklohmann) [inline coding-style.md]: This header is outdated.

File: coding-style.md [hunk 13]
```diff
@@ -0,0 +1,552 @@
...
+   *                   | | / / _ | / __/_  __/     Visibility                   *
+   *                   | |/ / __ |_\ \  / /          Across                     *
+   *                   |___/_/ |_/___/ /_/       Space and Time                 *
+   *                                                                            *
+   * This file is part of VAST. It is subject to the license terms in the       *
+   * LICENSE file found in the top-level directory of this distribution and at  *
+   * http://vast.io/license. No part of VAST, including this file, may be       *
+   * copied, modified, propagated, or distributed except according to the terms *
+   * contained in the LICENSE file.                                             *
+   ******************************************************************************/
```
Comment (dominiklohmann) [inline coding-style.md]: This header is outdated.

File: workflow.md [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+    `git rebase --interactive $(git merge-base origin/master HEAD)`.
+
+  + Create a pull request to `master` on GitHub.
+
+  + Wait for the results of continuous integration tools and fix any reported
+    issues.
+
+  + Ask a maintainer to review your work when your changes merge cleanly. If
+    you don't want a specific maintainer's feedback, ask for a review from the
+    `tenzir/backend` or `tenzir/frontend` teams.
```
Comment (dominiklohmann) [inline workflow.md]: These teams no longer exist; it's tenzir/engineering for everyone, or tenzir/vast, tenzir/devops, and tenzir/secops for more specific requests.

File: workflow.md [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+
+  + Address the feedback articulated during the review.
+
+  + A maintainer will merge the topic branch into `master` after it passes the
+    code review.
+
+- Similarly, for features or fixes relating to a specific GitHub issue, use
+  *issue branches* that branch off `master` with a naming convention of
+  `issue/XXX`, replacing XXX with the issue number.
```
Comment (dominiklohmann) [inline workflow.md]: Our branch protections do not support `issue/*`; instead `topic/*` should be used.

File: workflow.md [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+
+When introducing a new dependency or updating an existing one, please consider
+the following tasks:
+
+- **CMake:** Determine whether the dependency is a public, private, or
+  interface depdency or existing targets.
+
+- **Nix**: Update the Nix environment to link against the proper dependency.
+
+- **Dockerfile**: Update `Dockerfile` and `Dockerfile_prebuilt` to reflect the
```
Comment (dominiklohmann) [inline workflow.md]: The prebuilt Dockerfile no longer exists.

File: workflow.md [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+  dependency changes.
+
+- **CI**: Update the dependencies in GitHub Actions workflow files, including
+  the Jupyter Notebook workflow.
+
+- **Homebrew**: Notify the Homebrew package maintainer about the dependency
+  changes.
+
+- **Debian**: Notify the Debian package maintainer about the dependency
+  changes.
```
Comment (dominiklohmann) [inline workflow.md]: Both of these should be removed.

**Review by dominiklohmann — APPROVED** [review]
Fine by me. I left some comments inline that we should address before making this part of the docs on vast.io.

File: coding-style.md (hunk 5 above)
Comment (mavam) [reply coding-style.md:183]: Yes, but I'm inclined to leave it here since you talk about an exception.

File: coding-style.md (hunk 11 above)
Comment (mavam) [reply coding-style.md]: Can you make a rephrasing suggestion? I don't know how this works.

**Pull request merged**